pages/[[...params]].tsx

```diff
@@ -0,0 +1,15 @@
+import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import { parseDate } from "../lib/time";
+import { Dashboard } from "../components/Dashboard";
+import { parseYear } from "../lib/time";
+
+const Home: NextPage = () => {
+  const router = useRouter();
+  const [yearParam, dateParam] = (router.query.params || []) as string[];
+  const year = parseYear(yearParam);
+  const date = parseDate(dateParam);
+  return <Dashboard year={year} date={date} />;
```

Comment:
I should name this variable `selectedDate` instead of `date`. It would be obvious that they are not semantically the same. Still, `year` can be deduced from `selectedDate` so it would be less error prone to expect only one of them.